test/run-path-bad-package-name.js

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env node
+
+/* Copyright 2017 Mozilla
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. */
+
+'use strict';
+
+// Polyfill Promise.prototype.finally().
+require('promise.prototype.finally').shim();
+
+const path = require('path');
+const spawn = require('child_process').spawn;
+const tap = require('tap');
+
+let exitCode = 0;
+
+new Promise((resolve, reject) => {
+  // Paths are relative to the top-level directory in which `npm test` is run.
+  const child = spawn('node', [ path.join('bin', 'cli.js'), 'run', 'test/hello-world-bad-package-name/' ]);
+  const outputRegex = /Error: Invalid name: "hello world"/;
+
+  // let totalOutput = '';
+  let totalError = '';
+
+  child.stdout.on('data', data => {
+    const output = data.toString('utf8');
+    console.log(output);
+    // totalOutput += output.trim();
+  });
+
+  child.stderr.on('data', data => {
+    const error = data.toString('utf8');
+    console.error(error);
+    totalError += error.trim();
+    // if (outputRegex.test(totalError) && !quitting) {
+    //   child.kill('SIGINT');
+    //   quitting = true;
+    // }
```

Comment:
I've made this change in 834ec3c1415916b9875d1c1f19c92752000bbeea. I also made a minor change to this function in a82416ed0e9ba54997721e8dfa853e469df3132e.